ckan/templates/snippets/add_dataset.html

```diff
@@ -1,7 +1,9 @@
 {# Adds 'Add Dataset' button #}
 
+{% set dataset_type = dataset_type if dataset_type else 'dataset' %}
+
 {% if group %}
     {% link_for _('Add Dataset'), controller='package', action='new', group=group, class_='btn btn-primary', icon='plus-square' %}
 {% else %}
-    {% link_for _('Add Dataset'), controller='package', action='new', class_="btn btn-primary", icon="plus-square" %}
+    {% link_for _('Add ' + dataset_type.title()), controller='package', action='new', named_route=dataset_type + '_new', class_='btn btn-primary', icon='plus-square' %}
```

Comment:
@amercader does this cause any problem with translations? Now collecting the strings won't pick up "Add Dataset" anymore, will it get removed from the template and po files? (That would be bad)